### ISSUE TITLE
An option to disable emoji suggestions

### ISSUE THREAD

**Issue by hatarist** [overtake/TelegramSwift#293]
It was added recently and is a bit of pain in the ass for some people.
For example, the ":3" smiley is pretty popular among my Russian social circle, I use it oftentimes. If I type a - certainly meaningless, but still - message that contains just a ":3", it shows a " 3️⃣" emoji in the suggestion bar aboe the message field. If I hit Enter, the ":3" will be replaced with the " 3️⃣".

I'd really appreciate an option to turn the suggestions off; just like we already do have an option to "automatically replace emojis"

**Comment by yhaskell** [comment]
An alternative would be to atleast not select any emoji by default when you type a colon. That is very annoying -- let it show me emoji but let me press enter and send my colon without automatically put 😍emoji in my work chat.

**Comment by Drakal** [comment]
This is a feature on the Windows/Linux/Old Mac version of the app, it is very confusing that this newer Mac version lacks this important option, this should be implemented immediately to maintain feature parity.

**Comment by Drakal** [comment]
Actually the feature looks to be in the app already, but some lunatic had it be disabled under indecipherable conditions!
#306

**Comment by yhaskell** [comment]
As much as I understand the sentiment, calling people lunatics is probably not the best way to behave in public. Also not productive

**Comment by callumb** [comment]
I think the answer to this would be to allow people to map their own or also the change a default mapping to what suits them. For example I use XD and xD a lot but comes out at 😆 and 😂 respectively which isn't what I mean by them! So being able to delete those mappings but keep the others - Which I do find helpful.

**Comment by josh-berry** [comment]
This feature also causes problems with keyboard editing of a draft message. For example (in Telegram 5.7 (181163) AppStore):

1. Type "Hi htere :3" into the message field
2. Realize you made a typo ("htere")
3. Hit the left arrow key repeatedly to go back to fix the typo
4. Keep hitting the left arrow key for a while until you realize the cursor isn't actually moving because the emoji suggestion box has appeared
5. Hit "Esc" (while cursing under your breath in your preferred language) to get the #%&* thing to go away
6. Actually fix the typo

One alternative to providing an option to get rid of it might be to require the user to press a key or key combo (e.g. Up or Cmd+Up) to explicitly switch focus to the suggestion box. That would at least make it less obtrusive or likely to interfere with normal editing operations.

**Comment by matthewmorek** [comment]
I'd like to plug in, as this relates to an issue I've been experiencing for the last two weeks or so when using traditional emoticons, i.e: `:)`, `;)`, `:P`, `:D`, etc.

#### Issue
The above emoticons used to be automatically replaced by relevant emojis, when I finish typing (sort of like in Slack). As of a few updates ago they are not, and the settings are nowhere in the "General" section where they used to be.

Instead, I'm getting an emoji picker, but once I continue typing past the colon it disappears and nothing gets replaced.

#### Expected result
```
Hey! This is awesome! 😄
```
#### What actually happens
```
Hey! This is awesome! :D
```

Is this the intended behaviour, or is it a regression? Because this used to work perfectly and now it does not. I'm on `5.9.3 (191685) Beta`.

**Comment by callumb** [comment]
> I'd like to plug in, as this relates to an issue I've been experiencing for the last two weeks or so when using traditional emoticons, i.e: `:)`, `;)`, `:P`, `:D`, etc.
> 
> #### Issue
> The above emoticons used to be automatically replaced by relevant emojis, when I finish typing (sort of like in Slack). As of a few updates ago they are not, and the settings are nowhere in the "General" section where they used to be.
> 
> Instead, I'm getting an emoji picker, but once I continue typing past the colon it disappears and nothing gets replaced.
> 
> #### Expected result
> ```
> Hey! This is awesome! 😄
> ```
> 
> #### What actually happens
> ```
> Hey! This is awesome! :D
> ```
> 
> Is this the intended behaviour, or is it a regression? Because this used to work perfectly and now it does not. I'm on `5.9.3 (191685) Beta`.

Sadly this is now a feature... Instead of making it something you could turn off they have done the cheap thing and remove it. Now being recommended changing and adding in MacOS shortcuts to give back this functionality! That isn't a fix it's a dirty dirty hack as it then effects every other program on your Mac and something you shouldn't have to do to get a feature that's on just about every other IM program.

**Comment by zweeltow** [comment]
Fixed i guess. 

![image](https://user-images.githubusercontent.com/50041341/82680237-a7863d80-9c54-11ea-8ee5-8bb2cb17efd6.png)

**Comment by matthewmorek** [comment]
@zweeltow Not exactly. This is incomplete, as it only replaces emotes with emojis upon sending the message, not upon typing it out and proceeding to the next chunk of text (whether by adding punctuation, a space, or sending a message).

**Comment by josh-berry** [comment]
@zweeltow I think you're confusing two different things here:
1. Replace emoji automatically (so ":)" becomes "😄")
2. Showing a UI popup with emoji suggestions on what to replace

This issue is about the second of these two things, not the first. The OP even says [emphasis added]:

> I'd really appreciate an option to turn the suggestions off; _just like we already do have an option to "automatically replace emojis"_

**Comment by Drakal** [comment]
Basically, pressing enter at the end of a message that ends in a typed emote should just send the message as is, not replace that emote with the first suggested emoji.

Current workaround for this bug is to create a habit of pressing space before enter.

**Comment by PeterHeja** [comment]
I just want to bump this, as I've run into @josh-berry 's (https://github.com/overtake/TelegramSwift/issues/293#issuecomment-530618894) problem multiple times under a minute moments ago, and I just don't think disabling a feature that actually can make user experience worse should be impossible.

**Comment by dannote** [comment]
Can they be disabled, at least when I intend to write a chunk of monospaced code after backtick? This is exceedingly terrible when I write in Ruby or Elixir, where semicolons are widely used.

**Comment by ybizeul** [comment]
I don't understand this issue doesn't have more traction. It is quite infuriating to type a colon then return just to find yourself self greeted with a 👍

**Comment by piratrik** [comment]
I'm here to add that I'm also so frustrated about this emoji suggestion bar for the exact same reason as @josh-berry @PeterHeja and @ybizeul.

That bar ruins productivity and is super frustrating. Please let us be able to disable it.

**Comment by ppbdrinker** [comment]
It is indeed somewhat aggravating to see emoticon hints that can't be turned off.

Not every telegram user is willing communicate with their peers using thumbnails. Humanity has invented alphabets that are encouraged to be used to construct words and sentences.

**Comment by d8vjork** [comment]
This will be cool but I always use it, tho in the last update of Telegram Swift is being showing wrong suggestions and being super slow to refresh the majority of time

**Comment by anthonyloprimo** [comment]
I want to add to this.  I see this issue still hasn't been updated.  I want to push that at LEAST disabling the auto-focus would be ideal.  I actually like the idea of emoji suggestions (similar to suggesting stickers based on emoji) and it's a *fantastic* UX feature, however making it auto-focus and thus intercept some keyboard inputs (namely "enter") is a problem.

I would love to see an option to customize how this toolbar appears and behaves, at the very least, a change in behavior so it doesn't intercept any keyboard inputs.
